<launch>

    <arg name="config_file" default="$(find demo_rgbd_new)/config/robocane_config.yaml"/>
    <arg name="vins_folder" default="$(find demo_rgbd_new)/result/"/>

  <node pkg="demo_rgbd_new" type="featureTracking_node" name="feature_tracking" output="screen">
    <param name="config_file" type="string" value="$(arg config_file)"/>
    <remap from="/cam0/color" to="/camera/color/image_rect_color"/>
  </node>
  <node pkg="demo_rgbd_new" type="depth_handler_2" name="depth_handler" output="screen">
    <param name="config_file" type="string" value="$(arg config_file)"/>
    <remap from="/cam0/depth" to="/camera/aligned_depth_to_color/image_raw"/>
  </node>

  <node pkg="demo_rgbd_new" type="vio_node" name="vio_node" output="screen">
    <param name="config_file" type="string" value="$(arg config_file)"/>
    <param name="vins_folder" type="string" value="$(arg vins_folder)"/>
  </node>

  <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find demo_rgbd_new)/launch/vio_demo.rviz"/>
  
</launch>
